src/main/java/spark/route/RouteOverview.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2011- Per Wendel
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package spark.route;
+
+import java.util.ArrayList;
+import static java.util.Collections.singletonList;
+import java.util.List;
+import spark.Request;
+import spark.Response;
+import static spark.Spark.get;
+import spark.utils.Wrapper;
+
+public class RouteOverview {
+
+    /**
+     * Enables a route overview (showing all the mapped routes)
+     * The overview is made available at "/debug/routeoverview/"
+     * Calling this method before any other route mapping
+     * bas been performed will initialize the Spark server
+     */
+    public static void enableRouteOverview() {
+        enableRouteOverview("/debug/routeoverview/");
+    }
+
+    /**
+     * Enables a route overview (showing all the mapped routes)
+     * The overview is made available at the provided path
+     *
+     * @param path the path to the route overview
+     *             Calling this method before any other route mapping
+     *             bas been performed will initialize the Spark server
+     */
+    public static void enableRouteOverview(String path) {
+        get(path, RouteOverview::createHtmlOverview);
+    }
+
+    // Everything below this point is either package private or private
+
+    static List<RouteEntry> routes = new ArrayList<>();
+
+    static void add(RouteEntry entry, Object wrapped) {
+
+        if (wrapped instanceof Wrapper) {
+            entry.target = ((Wrapper) wrapped).delegate();
+        }
+
+        routes.add(entry);
+    }
+
+    static String createHtmlOverview(Request request, Response response) {
```

Comment:
Would love it if this was public so it can be used with instances like so:
```
Service sparkService = Service.ignite();
sparkService.get("/api", RouteOverview::createHtmlOverview);
```
instances are outlined in:
https://github.com/perwendel/spark/issues/297